pkg/kubelet/pluginmanager/operationexecutor/operation_generator.go

```diff
@@ -134,34 +141,31 @@ func (og *operationGenerator) GenerateRegisterPluginFunc(
 
 func (og *operationGenerator) GenerateUnregisterPluginFunc(
 	socketPath string,
-	pluginHandlers map[string]cache.PluginHandler,
+	pathToHandlers *cache.MapHandlerWithName,
 	actualStateOfWorldUpdater ActualStateOfWorldUpdater) func() error {
 
 	unregisterPluginFunc := func() error {
-		client, conn, err := dial(socketPath, dialTimeoutDuration)
-		if err != nil {
-			return fmt.Errorf("UnregisterPlugin error -- dial failed at socket %s, err: %v", socketPath, err)
+		var handlerWithName cache.HandlerWithName
+		handlerFound := false
+		pathToHandlers.Lock()
+		if len(pathToHandlers.Handlers) != 0 {
```

Comment:
if the handlers map is empty our behavior here is to just throw the error later because `handlerFound` is false. Why not just throw the error explicitly here instead of waiting for the knock on error later

Reply:
The error return on line 158 directly follows this.
I don't see much difference between returning error here versus current formation.